dts: arm: nxp: rt: Add dedicated rt1064 dtsi

loicpoulain opened the pull request:
Create rt1064 dtsi and add ocram1 node

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

MaureenHelm commented on dts/arm/nxp/nxp_rt1064.dtsi:
```diff
@@ -0,0 +1,14 @@
+#include <nxp/nxp_rt.dtsi>
+
+&flexram0 {
+	ocram1: ocram@20280000 {
```
RT1060 needs the same change. The difference between RT1060 and RT1064 is that the latter adds internal flash.

@galak, should we handle this with separate `nxp_rt1060.dtsi` and `nxp_rt1064.dtsi` files, or combine into `nxp_rt106x.dtsi`? I'm leaning toward combined.

galak replied:
What the difference between RT1060 & RT1064.  I'm probably same combinded, and at some point we can introduce an SoC set if needed.

MaureenHelm reviewed (changes requested):
I like the flexspi flash change, but I'd like to see the ocram added to rt1060 too. I think we could do this with another layer of dtsi includes:

`nxp_rt1060.dtsi`: includes `nxp_rt.dtsi` and adds the ocram
`nxp_rt1064.dtsi`: includes `nxp_rt1060.dtsi` and adds the internal flash

Commit:
dts: arm: nxp: rt: Create dedicated rt1060 dtsi

i.MX-RT1060 has 512KB additional on-chip RAM mapped at 0x20280000.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

Commit:
dts: arm: nxp: rt: Create dedicated rt1064 dtsi

The i.MXRT1064 inheriting from i.MXRT1060, has additional embedded
4-MB QSPI flash (via flexspi1).

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

loicpoulain commented:
> I like the flexspi flash change, but I'd like to see the ocram added to rt1060 too. I think we could do this with another layer of dtsi includes:
> 
> `nxp_rt1060.dtsi`: includes `nxp_rt.dtsi` and adds the ocram
> `nxp_rt1064.dtsi`: includes `nxp_rt1060.dtsi` and adds the internal flash

Done!

Commit:
dts: arm: nxp: rt106x: Unified OCRAM node

The iMX RT1060 and RT1064 have additional dedicated 512KB on-chip ram.
This OCRAM2 is mapped at 0x20200000, formerly OCRAM1 (flexram) mapping
which is moved to 0x20280000 in order to guarentee global OCRAM memory
continuity regardless OCRAM1 size configuration (256KB by default).

In default configuration, this gives 768KB (512+256) on-chip ram:
0x20200000 to 0x202BFFFF.

OCRAM2:           0x20200000 - 0x2027FFFF
OCRAM1(FlexRam):  0x2028FFFF - 0x202BFFFF

Add this memory region as a single node in the rt1060 device tree.

Note: MPU expects power of two memory region, in case of 768KB, let
the MPU configure 1MB instead.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>
